packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -219,6 +256,66 @@ - (UIImage *)scaleImage:(UIImage *)image by:(id)scaleParam {
   return image;
 }
 
+/**
+ * Creates a scaled version of the provided UIImage based on a specified scale factor. If the scale
+ * factor is significantly different from the image's current scale, a new UIImage object is
+ * created with the specified scale. Otherwise, the original image is returned.
+ *
+ * @param image The UIImage to scale.
+ * @param scale The factor by which to scale the image.
+ * @return UIImage Returns the scaled UIImage.
+ */
+- (UIImage *)scaleImage:(UIImage *)image withScale:(CGFloat)scale {
+  if (fabs(scale - image.scale) > 1e-3) {
+    return [UIImage imageWithCGImage:[image CGImage]
+                               scale:scale
+                         orientation:(image.imageOrientation)];
+  }
+  return image;
+}
+
+/**
+ * Scales an input UIImage to a specified size. If the aspect ratio of the input image
+ * is similar to the target size, the image's scale property is updated rather than resizing the
+ * image. If the aspect ratios significantly differ, the method redraws the image at the target
+ * size.
+ *
+ * @param image The UIImage to scale.
+ * @param size The target CGSize to scale the image to.
+ * @return UIImage Returns the scaled UIImage.
+ */
+- (UIImage *)scaleImage:(UIImage *)image toSize:(CGSize)size {
+  if (fabs(((int)image.size.width * image.scale) - size.width) > 0 ||
+      fabs(((int)image.size.height * image.scale) - size.height) > 0) {
+    if (fabs(image.size.width / image.size.height - size.width / size.height) < 1e-2) {
+      // Scaled image has close to same aspect ratio, updating image scale instead of resizing
+      // image.
+      return [self scaleImage:image
+                    withScale:(image.scale * (size.width / (image.size.width * image.scale)))];
+    } else {
+      UIGraphicsBeginImageContextWithOptions(size, NO, 1.0);
+      [image drawInRect:CGRectMake(0, 0, size.width, size.height)];
+      UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+      UIGraphicsEndImageContext();
+      // Return image with proper scaling
+      return [self scaleImage:newImage withScale:image.scale];
+    }
+  }
+  return image;
+}
+
+/**
+ * Scales the input CGSize by a specified scale factor and truncates the floating point values to
+ * integers.
+ *
+ * @param size The CGSize to scale.
+ * @param scale The scale factor to apply to the width and height of the CGSize.
+ * @return CGSize Returns the scaled CGSize with width and height as integers.
+ */
+- (CGSize)scaleSizeToInt:(CGSize)size withFactor:(CGFloat)scale {
+  return CGSizeMake((int)(size.width * scale), (int)(size.height * scale));
+}
+
```

Comment:
It looks like I forgot to flag it in the previous round, but the changes in this file also need native unit tests.